Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/TelemetryConfigurationFactoryTest.cs

```diff
@@ -915,6 +916,35 @@ public void LoadPropertiesGivesPrecedenceToValuesFromElementsBecauseTheyAppearBe
             Assert.AreEqual(42, instance.Int32Property);
         }
 
+        [TestMethod]
+        [ExpectedExceptionWithMessage(typeof(ArgumentException), "Failed to parse configuration value. Property: 'IntegerProperty' Reason: Input string was not in a correct format.")]
+        public void LoadPropertiesThrowsExceptionWithPropertyName()
+        {
+            // parsing this integer will throw "System.FormatException: Input string was not in a correct format."
+            // This is not useful without also specifying the errant property name.
+
+            XElement definition = XDocument.Parse(Configuration(
+                @"<TelemetryChannel Type=""" + typeof(StubTelemetryChannel).AssemblyQualifiedName + @""">
+                    <IntegerProperty>123a</IntegerProperty>
+                 </TelemetryChannel>")).Root;
+
+            var instance = new TelemetryConfiguration();
+
+            TestableTelemetryConfigurationFactory.LoadProperties(definition, instance, null);
+        }
+
+        [TestMethod]
```

Comment:
@cijothomas This is the extra test you asked for. :)
If this looks good to you, I'm ready to merge.